src/Microsoft.AspNet.Mvc.ModelBinding/Binders/KeyValuePairModelBinder.cs

```diff
@@ -40,7 +40,6 @@ public sealed class KeyValuePairModelBinder<TKey, TValue> : IModelBinder
             {
                 var untypedModel = propertyBindingContext.Model;
                 var model = ModelBindingHelper.CastOrDefault<TModel>(untypedModel);
-                parentBindingContext.ValidationNode.ChildNodes.Add(propertyBindingContext.ValidationNode);
```

Comment:
Would this have avoided looking at the `KeyValuePair` properties which we'd now start looking at? (since this does not have `ValidateAllProperties`)